### PR TITLE
[TBL] Call party direct when resending verification email

### PR DESCRIPTION
**Pull request by jcox-dev**
# Motivation and Context

Removes the call to backstage for resending verification email.

# What has changed

Changes the `resend_verification_email` method in reporting_units_controllers.py to hit the party service directly.

# How to test?

Run the unit tests, make sure they pass.
Tail the logs for backstage, party service. Hit the `/reporting-units/resend_verification/<ru_ref>/<party_id>` page. Make sure backstage doesn't get the call, but party service does.

### CODE REVIEW

**Comment by jcox-dev** [comment]
[response-operations-ui should directly call other services (not via backstage)](https://trello.com/c/0lPdzRbt/144-response-operations-ui-should-directly-call-other-services-not-via-backstage)

File: response_operations_ui/controllers/reporting_units_controllers.py [hunk 1]
```diff
@@ -56,8 +56,8 @@ def generate_new_enrolment_code(collection_exercise_id, ru_ref):
 
 def resend_verification_email(party_id):
     logger.debug('Re-sending verification email', party_id=party_id)
-    url = f'{app.config["BACKSTAGE_API_URL"]}/v1/reporting-unit/resend-verification-email/{party_id}'
-    response = requests.post(url)
+    url = f'{app.config["PARTY_URL"]}/party-api/v1/resend-verification-email/{party_id}'
+    response = requests.get(url, auth=app.config['PARTY_AUTH'])
```
Comment (tej99) [inline response_operations_ui/controllers/reporting_units_controllers.py]: Out of scope for this change but a get action feels wrong for this api in party. Possibly need to raise a TBL to change this?

File: response_operations_ui/controllers/reporting_units_controllers.py [hunk 2]
```diff
@@ -56,8 +56,8 @@ def generate_new_enrolment_code(collection_exercise_id, ru_ref):
...
 
     if response.status_code != 200:
```
Comment (tej99) [inline response_operations_ui/controllers/reporting_units_controllers.py]: Maybe change to raise_for_status if that's the pattern we're following in front stage

File: response_operations_ui/controllers/reporting_units_controllers.py (hunk 1 above)
Comment (jcox-dev) [reply response_operations_ui/controllers/reporting_units_controllers.py]: @tej99 tech debt item created for this to be considered 👍

**Pull request merged**